### PR TITLE
Add Makefile targets for some common/useful development tasks:

### LINKED ISSUE #357: Add standard make targets for local python development
As a developer, it would be nice if the "local setup" were more streamlined, so that we don't have to keep switching back and forth between the [Contributing](https://docs.jointakahe.org/en/latest/contributing/) page and a terminal.

I typically (and would suggest, minimally) creating  `Makefile` targets for:

- `'venv` - create the virtual environment (includes installing dependencies though sometimes I break those out into their own targets too)
- `.git/hooks/pre-commit` - install the pre-commit hooks for development
- `test` - run the pytests

I also like to wrap these in useful general targets:

- Something like `pydev` - create the virtual environment, install pre-commit hooks, etc
- Something like `startdb` to get the DB up and running (not always, it's sometimes useful)
- Targets to run different services

### PR DESCRIPTION
**Pull request by sivy**
## Development Make Targets

This adds several new `Makefile` targets to help streamline local development.

One decision I made on this was to use the docker-compose postgres rather than assuming a locally-hosted Postgres. Might not be everyone's choice.

### dev setup tasks:

- `.venv`
- /git/hooks/pre-commit
- .env

### db tasks: reuses the docker db container rather than hosting postgres locally:

- startdb
- stopdb
- migrations

### active development tasks:

- server
- stator

## Usage/Example

```
$ make pydev
python3 -m venv .venv
. .venv/bin/activate
python3 -m pip install -r requirements-dev.txt
[snip install stuff...]
Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.1 Pygments...
[snip install stuff...]
python3 -m pre_commit install
pre-commit installed at .git/hooks/pre-commit
```

```
make server
docker compose -f docker/docker-compose.yml up db -d
[+] Running 1/0
 ⠿ Container docker-db-1  Running                                                                                                        0.0s
python3 -m manage runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
January 05, 2023 - 16:33:20
Django version 4.1.5, using settings 'takahe.settings'
Starting development server at http://127.0.0.1:8000
```

Closes #357

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,3 +14,52 @@ compose_up:
...
+
+_PHONY: startdb stopdb
+startdb:
+	docker compose -f docker/docker-compose.yml up db -d
+
+stopdb:
+	docker compose -f docker/docker-compose.yml stop db
+
+_PHONY: superuser
+superuser: .env python-dev startdb
```
Comment (andrewgodwin) [inline Makefile:46]: Let's call this createsuperuser

File: Makefile [hunk 2]
```diff
@@ -14,3 +14,52 @@ compose_up:
...
+.env:
+	# afterwords, set in .env
+	# TAKAHE_DATABASE_SERVER="postgres://postgres:insecure_password@localhost:5433/takahe"
+	cp development.env .env
+
+_PHONY: python-dev
+python-dev: .venv
+
+_PHONY: precommit
+precommit: python-dev .git/hooks/pre-commit
```
Comment (andrewgodwin) [inline Makefile:36]: Why is this `python-dev`?

File: Makefile [hunk 3]
```diff
@@ -14,3 +14,52 @@ compose_up:
...
+_PHONY: test
+test: pydev
+	python3 -m pytest
+
+# Active development
+_PHONY: migrations server stator
+migrations: startdb
+	python3 -m manage migrate
+
+server: startdb
```
Comment (andrewgodwin) [inline Makefile:61]: Can these be `runserver` and `runstator` instead? I like makefile actions to sound more like what they're doing.

File: Makefile [hunk 4]
```diff
@@ -14,3 +14,52 @@ compose_up:
...
+# Development Setup
+.venv:
+	python3 -m venv .venv
+	. .venv/bin/activate
+	python3 -m pip install -r requirements-dev.txt
+
+.git/hooks/pre-commit: .venv
+	python3 -m pre_commit install
+
+.env:
```
Comment (andrewgodwin) [inline Makefile:27]: Can we combine this into the venv one and call it `setup_local` or something? It's a bit too granular right now.

File: Makefile (hunk 2 above)
Comment (sivy) [reply Makefile:36]: Sorry switched from `python-dev` to `pydev` halfway through hashtag-adhd

File: Makefile (hunk 3 above)
Comment (sivy) [reply Makefile:61]: sure thing

File: Makefile (hunk 2 above)
Comment (sivy) [reply Makefile]: Done

File: Makefile (hunk 4 above)
Comment (sivy) [reply Makefile:27]: I wouldn't mind something like:

```
_PHONY: setup_local
setup_local: .venv .env .git/hooks/pre-commit
```

And remove the current `precommit` target, then make everything else (that needs to) depend on `setup_local`. I do like to keep targets for specific files when there are actual files to be built.

Thoughts?

Comment (sivy) [reply Makefile:27]: I've made these changes, as well as switching the `development.env` template file to use the docker-hosted postgres by default.

**Review by andrewgodwin — APPROVED** [review]

**Comment by andrewgodwin** [comment]
Thanks for these - do you want to take a stab at writing the documentation for this as well?

**Comment by sivy** [comment]
Sure, in `contributing.md`?

**Comment by andrewgodwin** [comment]
No, in `docs/contributing.rst` if you could - that's where our detailed docs for local dev live.

**Comment by sivy** [comment]
Yeah that's what I meant, was writing from (bad) memory.

**Pull request merged**